# This patch file was generated by NetBeans IDE
# It uses platform neutral UTF-8 encoding and \n newlines.
+++ /var/www/agenlad/vendor/bcosca/fatfree/lib/auth.php
@@ -28,9 +28,27 @@
 		//! Mapper object
 		$mapper,
 		//! Storage options
-		$args;
+		$args,
+        //! DN del usuario LDAP
+        $dn;
 
 	/**
+     * Get auth::dn
+     * @return string
+     */
+    public function getDN(){
+        return $this->dn;
+    }
+    
+    /**
+     * Set auth::dn
+     * @param string $dn
+     */
+    public function setDN($dn){
+        $this->dn = $dn;
+    }
+    
+	/**
 	*	Jig storage handler
 	*	@return bool
 	*	@param $id string
@@ -119,7 +137,9 @@
 			($info=ldap_get_entries($dc,$result)) &&
 			@ldap_bind($dc,$info[0]['dn'],$pw) &&
 			@ldap_close($dc)) {
-			return $info[0]['uid'][0]==$id;
+                       $this->setDN($info[0]['dn']);
+			return $info[0]['uid'][0]==$id;
 		}
 		user_error(self::E_LDAP);
 	}
